presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -225,11 +225,49 @@ public void testWithDifferentFilterCondition()
-    public void testWithNoGroupByInBaseQuery()
+    public void testWithGroupBy()
     {
-        String originalViewSql = format("SELECT SUM(a) as sum_a, b FROM %s GROUP BY b", BASE_TABLE_1);
-        String baseQuerySql = format("SELECT b FROM %s", BASE_TABLE_1);
+        String originalViewSql = format("SELECT SUM(a) AS a, SUM(b*c) AS bc, d, e FROM %s GROUP BY d, e", BASE_TABLE_1);
+        String baseQuerySql = format("SELECT SUM(a) FROM %s", BASE_TABLE_1);
+        String expectedRewrittenSql = format("SELECT SUM(a) FROM %s", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT SUM(b*c) FROM %s WHERE d > 10", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT SUM(bc) FROM %s WHERE d > 10", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT SUM(a), d FROM %s GROUP BY d", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT SUM(a), d FROM %s GROUP BY d", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT SUM(a), SUM(b*c), d FROM %s GROUP BY d", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT SUM(a), SUM(bc), d FROM %s GROUP BY d", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT SUM(a), SUM(b*c), d, e FROM %s GROUP BY d, e", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT SUM(a), SUM(bc), d, e FROM %s GROUP BY d, e", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT SUM(a), SUM(b*c) FROM %s WHERE d IN (1, 2) AND IF(e, 1, 0) = 1", BASE_TABLE_1);
+        expectedRewrittenSql = format("SELECT SUM(a), SUM(bc) FROM %s WHERE d IN (1, 2) AND IF(e, 1, 0) = 1", VIEW);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+
+        baseQuerySql = format("SELECT d, e FROM %s", BASE_TABLE_1);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+
+        baseQuerySql = format("SELECT SUM(d) FROM %s GROUP BY e", BASE_TABLE_1);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+
+        baseQuerySql = format("SELECT SUM(a) FROM %s WHERE x > 10", BASE_TABLE_1);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+
+        baseQuerySql = format("SELECT SUM(a), x FROM %s GROUP BY x", BASE_TABLE_1);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
+
+        baseQuerySql = format("SELECT SUM(a) FROM %s WHERE f IN (1, 2)", BASE_TABLE_1);
+        assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
 
+        baseQuerySql = format("SELECT SUM(a) FROM %s WHERE IF(f, 1, 0) = 1", BASE_TABLE_1);
         assertOptimizedQuery(originalViewSql, baseQuerySql, baseQuerySql);
```

Comment:
Please add a test for
view: 
select SUM(a) from t where b > 0 group by c

base query: 
select sum(a) from t group by c

Reply:
If it is for testing the case that the view has `WHERE` while the query doesn't, I think it can be covered by the following right?

https://github.com/prestodb/presto/blob/ec973c63864d56b8f62a513b9786c3d8953c0ae9/presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java#L221-L224

Reply:
It is for testing the case where there are both group by and filters. Is there an existing test covering this?

It would also be good to add a nested `group by` case like below if there is no existing one. It is ok to not match the view. Just need to add a regression test to make sure it does not incorrectly match the view by some future changes.

view: select SUM(a), b, c from t group by b, c
base: select MAX(d) FROM (select SUM(a) d, b, c from t group by b, c)